quinn-proto/src/connection/mod.rs

```diff
@@ -4339,10 +4347,20 @@ impl Connection {
                         Some(cid_state) => {
                             let allow_more_cids = cid_state
                                 .on_cid_retirement(sequence, self.peer_params.issue_cids_limit())?;
+                            let path_id = path_id.unwrap_or_default();
+
+                            // If the path has closed, we do not issue more CIDs for this path
+                            // For details see  https://www.ietf.org/archive/id/draft-ietf-quic-multipath-17.html#section-3.2.2
+                            // > an endpoint SHOULD provide new connection IDs for that path, if still open, using PATH_NEW_CONNECTION_ID frames.
+                            let has_path = self.paths.get(&path_id).is_some();
```

Comment:
if we have some state, it does not mean it's open. The path could be abandoned already (we drop the state after some time) so we also need to check `abandoned_paths`